src/plone/restapi/serializer/atcontent.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+from Acquisition import aq_inner
+from Acquisition import aq_parent
+from Products.Archetypes.interfaces import IBaseFolder
+from Products.Archetypes.interfaces import IBaseObject
+from plone.restapi.interfaces import IFieldSerializer
+from plone.restapi.interfaces import ISerializeToJson
+from zope.component import adapter
+from zope.component import queryMultiAdapter
+from zope.interface import Interface
+from zope.interface import implementer
+
+
+@implementer(ISerializeToJson)
+@adapter(IBaseObject, Interface)
+class SerializeToJson(object):
+
+    def __init__(self, context, request):
+        self.context = context
+        self.request = request
+
+    def __call__(self):
+        result = {
+            '@context': 'http://www.w3.org/ns/hydra/context.jsonld',
+            '@id': self.context.absolute_url(),
+            '@type': self.context.portal_type,
+            'parent': {
+                '@id': aq_parent(aq_inner(self.context)).absolute_url(),
+                'title': aq_parent(aq_inner(self.context)).title,
+                'description': aq_parent(aq_inner(self.context)).description
+            },
+            'UID': self.context.UID(),
+        }
+
+        obj = self.context
+        for field in obj.Schema().fields():
+
+            if 'r' not in field.mode or not field.checkPermission('r', obj):
+                continue
+
+            name = field.getName()
+
+            serializer = queryMultiAdapter(
+                (field, self.context, self.request),
+                IFieldSerializer)
+            if serializer is not None:
+                result[name] = serializer()
+
+        return result
+
+
+@implementer(ISerializeToJson)
+@adapter(IBaseFolder, Interface)
+class SerializeFolderToJson(SerializeToJson):
+
+    def __call__(self):
+        result = super(SerializeFolderToJson, self).__call__()
+        result['member'] = [
+            {
+                '@id': member.absolute_url(),
+                'title': member.title,
+                'description': member.description
```

Comment:
The "BaseUnit" error explained in the comment is thrown here, it looks like "member.description" is a BaseUnit instance for Collage types and I don't know if for others too...

Perhaps this two title and descripton can be changed with something like

```python
 'title': member.getField('title').get(member), 
 'description': member.getField('description').get(member)
```

or 

```python
 'title': member.Title(), 
 'description': member.Description()
```

just guessing.